src/shogun/lib/SGMatrix.h

```diff
@@ -130,6 +131,15 @@ template<class T> class SGMatrix : public SGReferencedData
 		virtual ~SGMatrix();
 
 #ifndef SWIG // SWIG should skip this part
+		/** Resize matrix
+		 * If the new matrix is larger, fill the new memory with 0.
+		 *
+		 * @param rows new row numbers
+		 * @param cols new column numbers
+		 * @return if resizing was successful
+		 */
+		bool resize_matrix(index_t rows, index_t cols);
+
 		/** Get a column vector
```

Comment:
We dont you just allocate the matrix in different size rather than resizing it in the first bit of code?